atc/engine/check_delegate.go

```diff
@@ -107,6 +121,29 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 	var lock lock.Lock = lock.NoopLock{}
 	if d.plan.IsPeriodic() {
 		for {
+			lastCheck, err := scope.LastCheck()
+			if err != nil {
+				return nil, false, err
+			}
+
+			if d.build.IsManuallyTriggered() { // if the check was manually triggered
```

Comment:
Please double check my logic here 😟  @aoldershaw, I changed the if statements from being setting `shouldRun` to true, to returning false. 

This is what I converted it from:
https://github.com/concourse/concourse/blob/b047d257f2531c119e15d627e3799d9d931bad03/atc/engine/check_delegate.go#L136-L152

Reply:
Why move acquire lock to be after interval check? I'm not sure if the logic change is correct without a deep thinking, but if this optimization is not related to this PR, I would suggest divide this change to a separate PR, so that people can put more attention on it.

Reply:
This actually is related because it fixes a bug that this PR introduced, I wrote a summary of it in the commit that adds it here https://github.com/concourse/concourse/pull/7176/commits/28982bddba3b1433ce4e345eecc0a56d7dda6aa0. Let me know if you have any more questions!